src/blocks/ActionBlock/PhoneLink.astro

```diff
@@ -0,0 +1,38 @@
+---
+import type { HTMLAttributes } from 'astro/types';
+import type { PhoneLinkFragment } from '@lib/datocms/types';
+
+type PhoneLinkProps = Pick<
+  PhoneLinkFragment,
+  'action' | 'phoneNumber' | 'text'
+>;
+export type Props = HTMLAttributes<'a'> & PhoneLinkProps;
+const { action, phoneNumber, text, ...props } = Astro.props;
+
+const formatPhoneNumber = (phoneNumber: string) => {
```

Comment:
This is a bit of a naive formatter. Could pull in the very popular [`libphonenumber-js` package](https://www.npmjs.com/package/libphonenumber-js) (has zero dependencies) to do the phone number formatting: